views/emails/newPost.blade.php

```diff
@@ -1,14 +1,7 @@
-Hey {!! $user->display_name !!}!
-
-{!! $blueprint->post->user->display_name !!} made a post in a discussion you're following: {!! $blueprint->post->discussion->title !!}
-
-To view the new activity, check out the following link:
-{!! app()->url() !!}/d/{!! $blueprint->post->discussion_id !!}/{!! $blueprint->post->number !!}
-
----
-
-{!! $blueprint->post->content !!}
-
----
-
-You won't receive any more notifications about this discussion until you're up-to-date.
+{!! $translator->trans('flarum-subscriptions.email.new_post.body', [
+'{recipient_username}' => $user->display_name,
+'{poster_username}' => $blueprint->post->user->display_name,
+'{title}' => $blueprint->post->discussion->title,
+'{url}' => $url->to('forum')->route('discussion', ['id' => $blueprint->post->discussion_id, 'near' => $blueprint->post->number]),
+'{content}' => $blueprint->post->content
```

Comment:
Doesn't this now prevent injecting HTML? I'm not sure Flarum uses HTML here, but my Wordpress extension uses the fact `$post->content` is inserted as RAW to have HTML.

The change might be desirable, but it will have effect on some extensions.

Reply:
Would the translation engine escape the HTML?

Reply:
Actually my statement might not be correct. Because that body is used as a plain text email. Only fof/pretty-mail does HTML at the time, so the HTML encoding is actually in pretty-mail probably.

It does not need to be done together, but at some point we should make our emails HTML as well. And this will probably require changing these translations to have each bit of text in a separate translation.

Reply:
Let's leave it like this then, and revisit HTML email at a later time? Or I suppose we could split it up now to make that simpler for translators down the line. I suppose we could do:

- Salutation
- Intro
- Post Summary
- Ending (blank for now)

Reply:
Keep as is.

Reply:
I'm in the split-them-up camp, because the texts can then be reused in HTML emails (ticket #241). But not going to insist if there's already a majority...

Reply:
To be completely honest, I lean a little bit towards splitting (although not sure of what exactly the split should be), but I'm fine with keeping it intact either. Most core emails aren't split.

Additionally, with HTML emails, we could introduce a second version that might contain similar / same text, and split that. That might provide even more customizability. No need to use the exact same text for both?

Reply:
Maybe, maybe not. But I guess that itself is a good point in itself: It's hard to make a good decision about how to split these up currently, so it's probably better to defer that decision. :smile:

Reply:
Might make sense to have Blade View component like objects that have some properties by default (forum name, url, recipient etc) and then allow them to be rendered into html or text?

Anyhow, defer is also my opinion.